Makefile

```diff
@@ -1,22 +1,31 @@
-.PHONY: pre build release image clean
+# See: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63
+# For a list of valid GOOS and GOARCH values
+# Note: these can be overriden on the command line e.g. `make PLATFORM=<platform> ARCH=<arch>`
+PLATFORM=$(shell go env GOOS)
+ARCH=$(shell go env GOARCH)
 
+ifeq ("$(PLATFORM)", "windows")
+bin=compose-unpacker.exe
+else
+bin=compose-unpacker
+endif
 dist := dist
-bin := $(shell basename $(CURDIR))
-image := portainer/compose-unpacker
+image := portainer/compose-unpacker:latest
+.PHONY: build release image clean download-binaries
 
-pre:
-	mkdir -pv $(dist) 
+release: build image
 
-build: pre
-	GOOS="$(shell go env GOOS)" GOARCH="$(shell go env GOARCH)" CGO_ENABLED=0 go build --installsuffix cgo --ldflags '-s' -o $(bin)
-	mv $(bin) $(dist)/
+download-binaries:
+	mkdir -pv $(dist)
+	@./setup.sh $(PLATFORM) $(ARCH)
 
-release: pre
-	GOOS="$(shell go env GOOS)" GOARCH="$(shell go env GOARCH)" CGO_ENABLED=0 go build -a --installsuffix cgo --ldflags '-s' -o $(bin)
-	mv $(bin) $(dist)/
+build: download-binaries
+	GOOS="$(shell go env GOOS)" GOARCH="$(shell go env GOARCH)" CGO_ENABLED=0 go build -a --installsuffix cgo --ldflags '-s' -o dist/$(bin)
 
-image: release
-	docker build -t $(image) .
+image: build
```

Comment:
Now the instructions in the `readme` file  do not match the `Makefile`.

Generally when you build via make you don't necessarily want to make everything by default, just our binary.
CI step can and does pass arguments so not sure why we want to make image refer to build.

Here are the current Readme instructions.

```
# To just compile it and get the binary generated in dist/
make build

# To compile and build a Docker image:
make image
```

Therefore I would make these changes to the Readme.

1) `make` (no args) will build the first argument, just our binary nothing else . Convenience for dev.
2) `make build` - same as `make` (currently downloads binaries even if you don't need to which is not useful)
3) `make image` - make and build everything, this needs to ensure everything is there before creating the image so must depend on everything

For CI, the platform team just needs to call these I think:

```
make PLATFORM="windows" ARCH="amd64" image
make PLATFORM="linux" ARCH="amd64" image
make PLATFORM="linux" ARCH="arm" image
make PLATFORM="linux" ARCH="arm64" image
make PLATFORM="darwin" ARCH="amd64" image
make PLATFORM="darwin" ARCH="arm64" image
```


Therefore suggest the Makefile should be changed to:

```
# See: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63
# For a list of valid GOOS and GOARCH values
# Note: these can be overriden on the command line e.g. `make PLATFORM=<platform> ARCH=<arch>`
PLATFORM=$(shell go env GOOS)
ARCH=$(shell go env GOARCH)

ifeq ("$(PLATFORM)", "windows")
bin=compose-unpacker.exe
else
bin=compose-unpacker
endif

dist := dist
image := portainer/compose-unpacker:latest
.PHONY: build image clean download-binaries

build:
	GOOS="$(PLATFORM)" GOARCH="$(ARCH)" CGO_ENABLED=0 go build -a --installsuffix cgo --ldflags '-s' -o dist/$(bin)

download-binaries:
	mkdir -pv $(dist)
	@./setup.sh $(PLATFORM) $(ARCH)

image: build download-binaries
	docker build -f build/$(PLATFORM)/Dockerfile -t $(image) .

clean:
	rm -rf $(dist)
	rm -rf .tmp
	-docker rmi $(image)

```